app/pages/patient/patientinfo.js

```diff
@@ -601,62 +575,71 @@ var PatientInfo = translate()(React.createClass({
       if (patientInfo.about) {
         formValues.about = patientInfo.about;
       }
-    }
 
+      if(patientInfo.firstName) {
+        formValues.firstName = patientInfo.firstName;
+      }
+
+      if(patientInfo.lastName) {
+        formValues.lastName = patientInfo.lastName;
+      }
```

Comment:
Why repeat this code ?

Reply:
Not sure if I get It but  patientInfo.firstName can be different from personUtils.firstName().
patientinfo is retrieved from seagull structure `{ "profile": {"patient":{"firstName":"John","lastName":"Doe"}}}`
 personUtils.firstName()  is retrieved from seagull structure `{ "profile":{"firstName":"John","lastName":"Doe"}}`